### PR TITLE
upgrade PT to 1.12

### PR DESCRIPTION
**Pull request by lxning**
## Description

Please read our [CONTRIBUTING.md](https://github.com/pytorch/serve/blob/master/CONTRIBUTING.md) prior to creating your first pull request.

Please include a summary of the feature or issue being fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #(issue)
#1762 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Feature/Issue validation/testing

Please describe the Unit or Integration tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [X] [mac_install.log](https://github.com/pytorch/serve/files/9192671/mac.log)
- [X] [mac_regression_test.log](https://github.com/pytorch/serve/files/9201535/reg.txt)
- [X] [linux-cpu_install.log](https://github.com/pytorch/serve/files/9193963/linux-cpu.txt)
- [X] [linux-cpu_regression_test.log](https://github.com/pytorch/serve/files/9202405/reg.txt)
- [X] [win_cu116.log](https://github.com/pytorch/serve/files/9204391/win_cu116.log)
- [X] [win_cu113.log](https://github.com/pytorch/serve/files/9204392/win_cu113.log)
- [X] [win-cpu.log](https://github.com/pytorch/serve/files/9204393/win-cpu.log)
- [X] [linux_cu102.log](https://github.com/pytorch/serve/files/9204428/cu102.log)
- [X] [linux_cu113.log](https://github.com/pytorch/serve/files/9204429/cu113.log)
- [X] [linux_cu116.log](https://github.com/pytorch/serve/files/9204430/cu116.log)
- [X] [linux_gpu_regression_test.txt](https://github.com/pytorch/serve/files/9204431/reg.txt)

## Checklist:

- [ ] Did you have fun?
- [ ] Have you added tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

### CODE REVIEW

File: test/pytest/test_data/dlrm_bs_1.json [hunk 1]
```diff
@@ -0,0 +1,75 @@
+{
```
Comment (chauhang) [inline test/pytest/test_data/dlrm_bs_1.json:1]: Why is the dlrm test file part of PT 1.12 upgrade PR?

File: test/pytest/test_data/dlrm_bs_2.json [hunk 2]
```diff
@@ -0,0 +1,142 @@
+{
```
Comment (chauhang) [inline test/pytest/test_data/dlrm_bs_2.json:1]: Why is the dlrm test file part of PT 1.12 upgrade PR?

File: test/pytest/test_example_scriptable_tokenzier.py [hunk 3]
```diff
@@ -5,19 +5,34 @@
 import sys
 import time
 from argparse import Namespace
+from pathlib import Path
 
 import pytest
 import requests
 import test_utils
 import torch
-from torchtext.models import RobertaClassificationHead
```
Comment (chauhang) [inline test/pytest/test_example_scriptable_tokenzier.py:13]: What is the reason for including the changes to scriptable tokenizers as part of this PR?

**Review by chauhang — CHANGES_REQUESTED** [review]
Please see comments inline. Further check what changes are needed for torch_cu102_windows.txt

File: test/pytest/test_example_scriptable_tokenzier.py (hunk 3 above)
Comment (mreso) [reply test/pytest/test_example_scriptable_tokenzier.py:13]: torchtext upgrade breaks bc so we needed to adjust the unit test

File: test/pytest/test_data/dlrm_bs_1.json (hunk 1 above)
Comment (mreso) [reply test/pytest/test_data/dlrm_bs_1.json:1]: I cherry picked the fix for the scriptable tokenizer from the dlrm example branch, will remove these

Comment (lxning) [reply test/pytest/test_data/dlrm_bs_1.json]: Due to [torchtext backward incompatible](https://github.com/pytorch/text/releases/tag/v0.13.0), [test_example_scriptable_tokenzier.py](https://github.com/pytorch/serve/blob/master/test/pytest/test_example_scriptable_tokenzier.py#L41), which was added a few weeks ago, failed. So @mreso pulled [these fixes](https://github.com/pytorch/serve/pull/1767/commits/ad605276eeaf129ea01222ad3a1fe9bfd9617489) from DLRM.

File: test/pytest/test_data/dlrm_bs_2.json (hunk 2 above)
Comment (lxning) [reply test/pytest/test_data/dlrm_bs_2.json]: same reason as the above.

File: test/pytest/test_example_scriptable_tokenzier.py (hunk 3 above)
Comment (lxning) [reply test/pytest/test_example_scriptable_tokenzier.py:13]: same reason as the above

**Comment by lxning** [comment]
> Please see comments inline. Further check what changes are needed for torch_cu102_windows.txt

PT1.12 does not support cu102 on windows

**Review by maaquib — APPROVED** [review]

**Review by mreso — APPROVED** [review]
LGTM

**Review by chauhang — APPROVED** [review]

**Comment by chauhang** [comment]
@lxning Please check the Codebuild errors, failing for all the cases

**Pull request merged**